mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java

```diff
@@ -0,0 +1,165 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Set;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.MaybeSender;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MailboxSessionUtil;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.extension.PreDeletionHook;
+import org.apache.james.mailbox.fixture.MailboxFixture;
+import org.apache.james.mailbox.inmemory.InMemoryMailboxSessionMapperFactory;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.store.CombinationManagerTestSystem;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.message.DefaultMessageWriter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import reactor.core.publisher.Flux;
+
+class DeletedMessageVaultHookTest {
+
+    private static final String ALICE_ADDRESS = "alice@james.com";
+    private static final String BOB_ADDRESS = "bob@james.com";
+
+    private CombinationManagerTestSystem testingData;
+    private MessageIdManager messageIdManager;
+    private MessageManager messageManager;
+    private MailboxSession session;
+    private Mailbox mailbox;
+    private Message mailContent;
+    private DeletedMessageVault messageVault;
+    private Date internalDate;
+
+    private CombinationManagerTestSystem createTestingData(Set<PreDeletionHook> preDeletionHooks) {
+        return MessageIdManagerWithPreDeletionHooksTestSystemProvider.createTestingData(preDeletionHooks);
+    }
+
+    private DeletedMessage getDeletedMessage(MessageId messageId) throws Exception {
+        return DeletedMessage.builder()
+            .messageId(messageId)
+            .originMailboxes(mailbox.getMailboxId())
+            .user(session.getUser())
+            .deliveryDate(ZonedDateTime.ofInstant(internalDate.toInstant(), ZoneOffset.UTC))
+            .deletionDate(ZonedDateTime.now())
+            .sender(MaybeSender.getMailSender(ALICE_ADDRESS))
+            .recipients(new MailAddress(BOB_ADDRESS))
+            .content(Throwing.supplier(() -> new ByteArrayInputStream(asBytes(mailContent))))
+            .hasAttachment(false)
+            .build();
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        session = MailboxSessionUtil.create(MailboxFixture.ALICE);
+
+        messageVault = mock(DeletedMessageVault.class);
```

Comment:
Why mocking the vault?

Reply:
I want to verify vault is called in `notifyDelete` and capture the param `deletedMessage` to compare.

Reply:
It seems to be the source of the failed tests on the CI though